test/sqllogictest/pretty.slt

```diff
@@ -37,3 +37,65 @@ SELECT pretty_sql('')
 
 query error db error: ERROR: expected exactly one statement
 SELECT pretty_sql(';')
+
+#statement ok
+#CREATE CONNECTION kafka_conn TO KAFKA (BROKER 'localhost:9092', SECURITY PROTOCOL PLAINTEXT);
+
+#query T multiline
+#SELECT pretty_sql(create_sql) FROM mz_connections WHERE name = 'kafka_cann'
+#----
+#EOF
+
+statement ok
+CREATE CONNECTION kafka_conn TO KAFKA (BROKER 'localhost:9092', SECURITY PROTOCOL PLAINTEXT);
+
+query T multiline
+SELECT pretty_sql(create_sql) FROM mz_connections WHERE name = 'kafka_conn'
+----
+CREATE CONNECTION materialize.public.kafka_conn TO KAFKA (BROKER = 'localhost:9092', SECURITY PROTOCOL = plaintext);
+EOF
+
+statement ok
+CREATE TABLE t (i INT)
+
+query T multiline
+SELECT pretty_sql(create_sql) FROM mz_tables WHERE name = 't'
+----
+CREATE TABLE materialize.public.t (i [s20 AS pg_catalog.int4]);
```

Comment:
The type here is ugly because `SHOW CREATE` runs an extra simplify step that depends on the catalog. I'm not sure yet how to implement that in sql. It might need to be an unmat fn, but it'd have to be unary and...return the whole catalog as a new type? Very unsure what to do.

Reply:
In `pack_update_*` we have access to the `CatalogState`, could we add a method like [simplify_names](https://github.com/MaterializeInc/materialize/blob/main/src/sql/src/plan/statement/show.rs#L893) that uses `CatalogState` instead of a `SessionCatalog`?

Reply:
That would record a string from a snapshot of a catalog. I'm concerned a rename would make that string incorrect. It is possible those renames already do the right thing (retract and insert new pack rows for dependents), but I'm not sure. I like keeping this column as the raw `to_ast_string_stable` output. I'd prefer the simplification happen as part of the call to `pretty_sql` to avoid any possible snapshot skew problems.

Reply:
>  I'm concerned a rename would make that string incorrect.

Yeah, I have the same concern.

> I'd prefer the simplification happen as part of the call to pretty_sql to avoid any possible snapshot skew problems.

Same. Making `pretty_sql` a materializable function seems _real_ hard but it does generally seem much safer for the `create_sql` column to be the output of `to_ast_string_stable`.